source/agora/common/Transaction.d

```diff
@@ -317,5 +345,86 @@ unittest
 
     // It isn't validated. (the sum of `Output` > the sum of `Input`)
     assert(!secondTx.verifyData(findOutput), format("Transaction data is not validated %s", secondTx));
+}
+
+/// This creates a new transaction and signs it as a publickey
+/// of the previous transaction to create and validate the input.
+unittest
+{
+    import std.format;
+
+    Transaction[Hash] storage;
+
+    immutable(KeyPair)[] key_pairs;
+    key_pairs ~= KeyPair.random();
+    key_pairs ~= KeyPair.random();
+    key_pairs ~= KeyPair.random();
+
+    // delegate for finding `Output`
+    auto findOutput = (Hash hash, size_t index)
+        {
+            if (auto tx = hash in storage)
+            {
+                if (index < tx.outputs.length)
+                {
+                    return &tx.outputs[index];
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            else
+            {
+                return null;
+            }
+        };
+
+    // Create the first transaction.
+    Transaction genesisTx = newCoinbaseTX(key_pairs[0].address, 100);
+    Hash genesisHash = hashFull(genesisTx);
+    storage[genesisHash] = genesisTx;
+
+    // Create the second transaction.
+    Input txIn =
+    {
+        previous : genesisHash,  // hash of previous utxo
+        index : 0  // first item in the outputs array
+    };
+
+    // Signs the previous hash value.
+    txIn.signature = key_pairs[0].secret.sign(hashFull(txIn)[]);
 
+    Transaction tx1 = Transaction(
+        [
+            txIn
+        ],
+        [
+            Output(10_000, key_pairs[1].address)
+        ]
+    );
+
+    assert(tx1.verifySignature(findOutput), format("Transaction signature is not validated %s", tx1));
+
+    Input txIn2 =
+    {
+        previous : genesisHash,  // hash of previous utxo
+        index : 0  // first item in the outputs array
+    };
+
+    // Sign with incorrect key
+    txIn2.signature = key_pairs[1].secret.sign(hashFull(txIn2)[]);
+
+    Transaction tx2 = Transaction(
+        [
+            txIn2
+        ],
+        [
+            Output(10_000, key_pairs[1].address)
```

Comment:
The Output of the previous transaction has 100.
I'd better give you a price smaller or equal to this